tests/phpunit/tests/db.php

```diff
@@ -2472,4 +2472,91 @@ public function data_parse_db_host() {
 			),
 		);
 	}
+
+	/**
+	 * @ticket 55414
+	 */
+	public function test_suppress_query_error() {
+		global $wpdb;
+
+		$old_show_errors     = $wpdb->show_errors;
+		$old_suppress_errors = $wpdb->suppress_errors;
+		$old_last_query      = $wpdb->last_query;
+		$old_error_log_path  = ini_get( 'error_log' );
+
+		$suppress_query_error_filter = new MockAction();
+		$log_query_error_action      = new MockAction();
+
+		try {
```

Comment:
Instead of `try/finally`, tests in the WP Core test suite typically follows this pattern:

```php
// Store the old value.
$old_value = $value;
// Change the value.
$value     = $new_value;
// Do something that uses value internally.
$actual    = do_the_thing();
// Restore the old value.
$value     = $old_value;
// Assert.
$this->assertSame( $expected, $actual, 'The thing did not return the expected value.' );

Reply:
Thank you for the guidance. This is now fixed.